rest_tools/server/handler.py

```diff
@@ -171,6 +171,19 @@ def prepare(self):
                 raise tornado.web.HTTPError(503, reason="server overloaded")
             self.start_time = time.time()
 
+        # decode JSON-encoded requests body
+        if self.request.body:
+            try:
+                self.request.body_arguments = json_decode(self.request.body)
+            except json.JSONDecodeError:
+                raise tornado.web.HTTPError(
+                    400, reason="requests body is not JSON-encoded"
+                )
+            if not isinstance(self.request.body_arguments, dict):
+                raise tornado.web.HTTPError(
+                    400, reason="JSON-encoded requests body must be a 'dict'"
+                )
```

Comment:
I would prefer not to do this in `prepare()` as that applies to all requests everywhere.  Maybe use a cached function in `arghandler.py`, so it's only ever done once?

Reply:
[`4491d41` (#138)](https://github.com/WIPACrepo/rest-tools/pull/138/commits/4491d4135b77be3adb0c8501cb240b88ea901ab2)